packages/react-events/src/__tests__/Focus-test.internal.js

```diff
@@ -258,17 +261,63 @@ describe('Focus event responder', () => {
       expect(onFocusChange).toHaveBeenCalledTimes(2);
       expect(onFocusChange).toHaveBeenCalledWith(false);
     });
+
+    it('is not called after "blur" and "focus" events on nested descendants', () => {
+      innerRef.current.dispatchEvent(createFocusEvent('focus'));
+      expect(onFocusChange).toHaveBeenCalledTimes(0);
+      innerRef.current.dispatchEvent(createFocusEvent('blur'));
+      expect(onFocusChange).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  describe('onFocusWithinChange', () => {
+    let onFocusWithinChange, ref, innerRef;
+
+    beforeEach(() => {
+      onFocusWithinChange = jest.fn();
+      ref = React.createRef();
+      innerRef = React.createRef();
+      const element = (
+        <Focus onFocusWithinChange={onFocusWithinChange}>
+          <div ref={ref}>
+            <a ref={innerRef} />
+          </div>
+        </Focus>
+      );
+      ReactDOM.render(element, container);
+    });
+
+    it('is called after "blur" and "focus" events immediate child', () => {
+      ref.current.dispatchEvent(createFocusEvent('focus'));
+      expect(onFocusWithinChange).toHaveBeenCalledTimes(1);
+      expect(onFocusWithinChange).toHaveBeenCalledWith(true);
+      ref.current.dispatchEvent(createFocusEvent('blur'));
+      expect(onFocusWithinChange).toHaveBeenCalledTimes(2);
+      expect(onFocusWithinChange).toHaveBeenCalledWith(false);
+    });
+
+    it('is called after "blur" and "focus" events on descendants', () => {
```

Comment:
There should be test coverage to ensure the callback is not called every time focus moves within the subtree